xrpl4j-integration-tests/src/test/java/org/xrpl/xrpl4j/tests/environment/RippledContainer.java

```diff
@@ -132,7 +133,11 @@ private void waitForLedgerTimeToSync() {
 
   private ZonedDateTime getLedgerTime() {
     try {
-      return getXrplClient().serverInfo().time();
+      return getXrplClient().serverInformation().info().map(
+        rippled -> rippled.time(),
+        clio -> null,
```

Comment:
We shouldn't return `null` from this method because it breaks the contract (i.e., `ZonedDateTime` isn't `Optional`.

That said, what if for `clio` and `reporting` we simply returned the host's time. There's no other time to get anyway, and up above where this method is used, the "wait until system time matches the host time" will be essentially reduced towards 0.

Reply:
Also, a unit test for this method should exist (might need to make the method `package-private` to pull that off.